src/twisted/python/compat.py

```diff
@@ -245,7 +245,11 @@ def execfile(filename, globals, locals=None):
     """
     if locals is None:
         locals = globals
-    with open(filename, "rbU") as fin:
+    if _PY3:
+        mode = "rb"
+    else:
+        mode = "rbU"
```

Comment:
Hang on, what the hell is this?

[The Python 2.7 documentation says](https://docs.python.org/2/library/functions.html#open) (note my emphasis):

> In addition to the standard fopen() values mode may be `'U'` or `'rU'`. Python is usually built with universal newlines support; supplying `'U'` **opens the file as a text file**, but lines may be terminated by any of the following: the Unix end-of-line convention `'\n'`, the Macintosh convention `'\r'`, or the Windows convention `'\r\n'`.

How the hell are we passing both binary mode *and* universal newline mode? This simply makes no sense. Given that the Python 3 version does not provide universal newlines, we clearly don't need them, so why don't we avoid introducing a branch altogether and just change the mode to `rb` unconditionally.